pipeline/load.py

```diff
@@ -31,42 +48,150 @@ def get_data_from_db(conn: connection, table: str, column = "*")-> pd.DataFrame:
     return pd.DataFrame(result)
 
 
-def check_for_duplicates(conn: connection, df: pd.DataFrame, table: str, column: str) -> pd.DataFrame:
-    """Checks a local dataframe against a database table on specific column.
-    Returns non-duplicative local data only"""
-    df = df.copy()
-    articles = get_data_from_db(conn, table)
-    df = df[not df[column].isin(articles[column])]
-    # test this line
+def check_for_duplicate_articles(conn: connection, url: str) -> str:
+    """Checks for a duplicates in article table. Returns a dataframe without duplicates."""
+
+    with conn.cursor() as cur:
+        cur.execute("""SELECT * FROM article where article_url = %s;""", [url])
+        duplicate = cur.fetchall()
+        if len(duplicate) > 0:
+            return None
+    return url
+
+
+def check_for_duplicate_authors(conn: connection, name: str) -> str:
+    """Checks for a duplicates in author table. Returns a dataframe without duplicates."""
 
-    return df
+    with conn.cursor() as cur:
+        cur.execute("""SELECT * FROM author where author_name = %s;""", [name])
+        duplicate = cur.fetchall()
+        if len(duplicate) > 0:
+            return None
+    return name
```

Comment:
Are these functions making sure we don't add articles with the same id twice or authors with the same name twice?
Could it be easier to make these columns unique in the DB schema?

Reply:
All looks v good though :)

Reply:
If we did a UNIQUE constraint, I think then we'd have to do a try:except loop and still log which articles were rejected - i think my way *might* be faster or the same

Reply:
The two functions each check for article and for author tables respectively also

Reply:
Couldn't we do a ON CONFLICT DO NOTHING statement like we did last week, do we need to log the rejections?

Reply:
okay cool idm either way, it's very minor